File: src/main/java/com/github/weewar/mapviewer/model/enums/TerrainType.java
package com.github.weewar.mapviewer.model.enums;

public enum TerrainType {
    airfield,
    bridge,
    city,
    desert,
    forest,
    harbor,
    mountain,
    plain,
    repairshop,
    swamp,
    water;

    public static TerrainType of(String str) {
        return str != null ? valueOf(str) : null;
    }
}

File: src/main/kotlin/com/github/weewar/mapviewer/model/enums/TerrainType.kt
package com.github.weewar.mapviewer.model.enums

enum class TerrainType {
    airfield,
    bridge,
    city,
    desert,
    forest,
    harbor,
    mountain,
    plain,
    repairshop,
    swamp,
    water;

    companion object {
        fun of(str: String): TerrainType = valueOf(str)
    }
}

